src/world.rs

```diff
@@ -429,31 +447,103 @@ impl LazyInsertions {
     /// struct InsertPos;
     ///
     /// impl<'a> System<'a> for InsertPos {
-    ///     type SystemData = (Entities<'a>, Fetch<'a, LazyInsertions>);
+    ///     type SystemData = (Entities<'a>, Fetch<'a, LazyUpdate>);
     ///
     ///     fn run(&mut self, (ent, lazy): Self::SystemData) {
     ///         let a = ent.create();
     ///         let b = ent.create();
     ///
-    ///         lazy.add(vec![(a, Pos(3.0, 1.0)), (b, Pos(0.0, 4.0))]);
+    ///         lazy.insert(vec![(a, Pos(3.0, 1.0)), (b, Pos(0.0, 4.0))]);
     ///     }
     /// }
     /// ```
-    pub fn add<L>(&self, l: L)
+    pub fn insert<L>(&self, l: L)
         where L: LazyInsert + 'static
     {
         self.stack.push(Box::new(l));
     }
+
+    /// Lazily removes a component.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// # use specs::*;
+    /// #
+    /// struct Pos;
+    ///
+    /// impl Component for Pos {
+    ///     type Storage = VecStorage<Self>;
+    /// }
+    ///
+    /// struct RemovePos;
+    ///
+    /// impl<'a> System<'a> for RemovePos {
+    ///     type SystemData = (Entities<'a>, Fetch<'a, LazyUpdate>);
+    ///
+    ///     fn run(&mut self, (ent, lazy): Self::SystemData) {
+    ///         for entity in ent.join() {
+    ///             lazy.remove::<Pos>(entity);
+    ///         }
+    ///     }
+    /// }
+    /// ```
+    pub fn remove<C>(&self, e: Entity)
+    where
+        C: Component + Send + Sync,
+    {
+        self.stack.push(Box::new(LazyRemovalInfo::<C>(e, PhantomData)));
```

Comment:
Actually, this could just use `execute` now. That way, there's no need for the removal struct.